controller/service.go

```diff
@@ -106,10 +108,31 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 			c.clearServiceState(key, svc)
 			lbIPs = []net.IP{}
 		}
-		// User set or changed the desired LB IP, nuke the
-		// state. allocateIP will pay attention to LoadBalancerIP and try
-		// to meet the user's demands.
-		if svc.Spec.LoadBalancerIP != "" {
+		// User set or changed the desired LB IP(s), nuke the
+		// state. allocateIP will pay attention to LoadBalancerIP(s) and try
+		// to meet the user's demands. annotationLoadBalancerIPs takes higher
+		// precedence over svc.Spec.LoadBalancerIP.
+		// TODO: remove check for svc.Spec.LoadBalancerIP when this field is removed
+		// from service spec.
+		desiredLbIPs, err := c.getDesiredLbIPs(svc)
+		if err != nil {
```

Comment:
What's missing here is a check on ipFamily. If the user puts the wrong ip family compared to the service's one, we must fail.
I'd suggest incorporating that check to have `getDesiredLbIPs` return a slice and always check ipFamily. That will work also with the loadbalancerip type.

Reply:
done